helixbass/tree-sitter-lint-lsp
# Basic LSP server setup

In this PR:
- try and get a basic LSP server "up and running"

To test:
You'd need to have helixbass/tree-sitter-lint#15 checked out in a sibling directory I believe (and per previous testing instructions on those PR's run `cargo build --release` in its `.tree-sitter-lint/tree-sitter-lint-local/` local "custom binary project" directory) (because this is currently hard-coding a local-path dependency on it)

At that point you should be able to eg `cargo build --release` and then (not entirely clear on what the requirements would be at the moment for the project directory in which you're running this but the `tree-sitter-lint/` repo directory would probably be a good bet) do whatever you have to do to tell your editor to launch it as an LSP server against a Rust file/project

Then in theory you should see ("readonly", no fixing yet) diagnostics for things like `Default::default()`
